galsim/chromatic.py

```diff
@@ -1226,6 +1226,60 @@ def __init__(self, original, waves, oversample_fac=1.0, use_exact_sed=True,
 
         # Don't interpolate an interpolation.  Go back to the original.
         self.deinterpolated = original.deinterpolated
+    
+    @classmethod
+    def from_images(cls, images, waves, _force_stepk = None, _force_maxk = None, **kwargs):
+        """
+        Alternative initiazliation to InterpolatedChromaticObject from input images at specific wavelenghts. Any parameters accepted by
+        the InterpolatedImage class can be passed in kwargs. Note that stepk and maxk are parameters that can depend on the image size,
+        and therefore on the wavelength. If not given, as a list for every image, or a single number for all images, it will be caluclated
+        using the input image pixel scale and dimensions. This means it will be identical for all images. This will cause small differences
+        from the normal use of this class using chromatic objects whose stepk and maxk are wavelength-dependant. 
+        
+        Parameters:
+            images:          list of Galsim Image objects to use as interpolated images.
+            wave:            list of wavelength values in nanometers.
+            _force_stepk:    list of step_k values to pass to InterpolatedImages for each image. Can also be single value
+                             to pass for all images alike. If not given stepk is calculated from image pixel scale and dimensions. [Default: None]
+            _force_maxk:     list of max_k values to pass to InterpolatedImages for each image. Can also be single value
+                             to pass for all images alike. If not given maxk is calculated from image pixel scale. [Default: None]
+
+        Returns:
+            An InterpolatedChromaticObject intialized from input images.
+        """
+        obj = cls.__new__(cls)  # Does not call __init__
+        obj.waves = np.array(waves) # images are assumed to be sorted by wavelength
```

Comment:
I think we should check whether they are actually sorted and raise an exception if they aren't.  Should also document this requirement in the docstring.

Reply:
Thanks for the suggestion! I I have added this main function do the sanity checks and then have it call the "leading underscore". I haven't pushed these changes yet due to a question about the other comment about the stepk, maxk.